src/tests/test_commandline.rs

```diff
@@ -40,3 +40,68 @@ fn commandline_test_replace() -> TestResult {
         456",
     )
 }
+
+#[test]
+fn commandline_test_cursor() -> TestResult {
+    run_test(
+        "commandline --replace '0😀2'\n\
```

Comment:
emoji's are great to test. I'm wondering if you can add a test with an emoji with one more more ZWJs? https://emojipedia.org/emoji-zwj-sequence/

I think it will help prove this solution is working.

Reply:
This would be a good one to try 👩‍❤️‍👩
You can see it here disassembled. It has `woman + zwj + heavy_black_heart + variation_selector + zwj + woman`
```
"👩‍❤️‍👩" | debug -r 
String {
    val: "👩\u{200d}❤\u{fe0f}\u{200d}👩",
    span: Span {
        start: 202072,
        end: 202094,
    },
}
```

Reply:
Thanks, great test! Changed my test to that now.